app/src/main/java/sistemas/puc/com/finantialapp/FetchRatesTask.java

```diff
@@ -90,4 +101,27 @@ protected void onProgressUpdate(Void... voids) {}
 
     @Override
     protected void onPostExecute(List<MoedaItem> result) {}
+
+    private List<MoedaItem> getRatesFromJson(String jsonStr) throws JSONException{
+
+        final String FIXER_RATES = "rates";
+
+        List<MoedaItem> result = new ArrayList<>();
+
+        JSONObject json = new JSONObject(jsonStr);
+        JSONObject ratesJson = json.getJSONObject(FIXER_RATES);
+
+        Iterator<String> rates = ratesJson.keys();
+
+        while (rates.hasNext()) {
+            String code = rates.next();
+            String name = "NameFromHash"; // TODO: create a hashtable to get the name from code.
+            double rate = ratesJson.getDouble(code);
+            // Temporarily using a placeholder Date before
+            // deciding if date should even be in MoedaItem.
+            result.add(new MoedaItem(code, name, rate, Util.getTimeFromDate(7,10,2017)));
```

Comment:
@tabocu Should each MoedaItem really contain the date? Or should I remove it from our data structure?

Reply:
Ass discussed, we will keep that way.